internal/graph/storagewrapper.go

```diff
@@ -344,12 +349,13 @@ func (c *CachedDatastore) newCachedIterator(
 		objectID:          objectID,
 		relation:          relation,
 		userType:          userType,
+		wg:                c.wg,
 	}, nil
 }
 
 // Close closes the datastore and cleans up any residual resources.
 func (c *CachedDatastore) Close() {
-	c.OpenFGADatastore.Close()
+	c.wg.Wait()
```

Comment:
⚠️ this will not work with my PR https://github.com/openfga/openfga/pull/2126. Because there won't be a Close() method anymore.